### PR TITLE
Add support for internal tables (aka managed tables)

### PR DESCRIPTION
**Pull request by jphalip**

### CODE REVIEW

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/utils/avro/AvroSerializer.java
Comment (davidrabinowitz) [inline connector/src/main/java/com/google/cloud/hive/bigquery/connector/utils/avro/AvroSerializer.java:34]: Can't we use built in AvroSerDe implementation?

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Missing unit tests, especially for the heavy logic classes as serializers/deserializers and schema convertors

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/BigQueryMetaHook.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/BigQueryMetaHook.java:275]: Yes it can, but that goes through a different workflow in `BigQueryOutputCommitter#commitJob(JobContext)`. This is mentioned in the method comment above.

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/JobInfo.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/JobInfo.java:37]: This file contains some information that might be needed at different stages of the job and from different environments (e.g. individual sub-tasks during the job's execution, and the main process at the end when committing the results). But now that you point that out, the HiveBigQueryConfig gets serialized and is accessible from all environments, is that right? If so, then yes, we could probably merge them.

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/utils/avro/AvroSerializer.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/utils/avro/AvroSerializer.java:34]: For the same reason as in PR #12, I had to write my own version to get around some data loss issues, i.e. the built-in AvroSerDe loses nano precision for timestamps...

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/BigQueryConstantDesc.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/BigQueryConstantDesc.java:44]: Yes will do

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/arrow/ArrowBatchReader.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/arrow/ArrowBatchReader.java:112]: Ok, should I wait for that release before merging this PR?

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/BigQueryInputSplit.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/BigQueryInputSplit.java:199]: Ok, I'll look into it

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/JobInfo.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/JobInfo.java]: Ok so for now I've renamed the class to `JobDetails`. As for merging that information with `HiveBigQueryConfig`, we can look into it later -- See this issue I've just created: https://github.com/GoogleCloudDataproc/hive-bigquery-connector/issues/22

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/BigQueryInputSplit.java
Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/BigQueryInputSplit.java]: Yes agreed. I think I'm probably not using the injector the right way. This can be improved later as part of https://github.com/GoogleCloudDataproc/hive-bigquery-connector/issues/22

**Comment by jphalip** [comment]
Closing as this code was merged with the rest of the `develop` branch into `main`.

**Pull request closed without merging**
